test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/Common/Command/QueueCommand.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Common;
+
+// Responsible for creation and execution of commands to interact with Queue Type resources(External Resources).
+public class QueueCommand : IExecutableCommand<QueueResponse>
+{
+	private readonly IQueueManager<QueueRequest, QueueResponse> _queueManager;
+	private readonly string _queueName;
+	private readonly QueueOperation _queueOperation;
+	private readonly QueueType _queueType;
+
+	public QueueCommand(QueueType queueType, QueueOperation queueOperation, string queueName)
+	{
+		_queueType = queueType;
+		_queueName = queueName;
+		_queueOperation = queueOperation;
+		if (QueueType.EventHub == queueType)
```

Comment:
L22-29 looks like a factory pattern approach shall we refactor this to a factory class ??

Reply:
Sure. We can take this up in the improvement task.

Reply:
I think it's a small task can be taken care in the same PR